common/server_room.cpp

```diff
@@ -36,6 +36,33 @@ Server_Room::~Server_Room()
     usersLock.unlock();
 }
 
+bool Server_Room::userMayJoin(const ServerInfo_User & userInfo)
+{
+    
+    ServerInfo_Room roomInfo;
```

Comment:
But that's a different function and this one isn't in scope there. Also it's a different type, this one is Server_Room and the other is ServerInfo_Room. I think you're mixed up here